mem_test.go

```diff
@@ -191,3 +202,29 @@ func BenchmarkHash(b *testing.B) {
 		}
 	}
 }
+
+// very old go like 1.14 doesn't have io.Discard,
+// but ioutil.Discard are deprecated after 1.16.
```

Comment:
Sorry, I could have been maybe more clear with a more direct example:
```
// ioutil.Discard was changed to io.Discard from go 1.16; so this is in part for compatibility.
// Also, we limit our implementation to io.Writer for more narrow and specific testing.
```

Reply:
yes, this looks better